tests/integration/scripts/test_validate_integration.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python3
+
+# THIS FILE IS PART OF THE CYLC WORKFLOW ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Integration tests for Cylc Validate CLI script."""
+
+from cylc.flow.scripts.validate import wrapped_main as validate
+from cylc.flow.parsec.exceptions import IllegalItemError
+import pytest
+from cylc.flow.parsec.exceptions import Jinja2Error
+
+
+async def test_revalidate_checks_source(
+    capsys, validate, workflow_source, install, one_conf
+):
+    """Validation fails if revalidating with broken config.
+    """
+    src_dir = workflow_source(one_conf)
+    workflow_id = install(src_dir)
+
+    # Check that the original installation validates OK:
+    validate(workflow_id, against_source=True)
+
+    # Break the source config:
+    with open(src_dir / 'flow.cylc', 'a') as handle:
+        handle.write('\n[runtime]\n[[foo]]\nAgrajag = bowl of petunias')
+
+    # # Check that Validate now fails:
+    with pytest.raises(IllegalItemError, match='Agrajag'):
+        validate(workflow_id, against_source=True)
+
+
+async def test_revalidate_gets_old_tvars(
```

Comment:
The "revalidate" terminology should be gone now (it'll be cryptic down the line):

```suggestion
async def test_validate_from_source(
    capsys, validate, workflow_source, install, one_conf
):
    """Validation fails if revalidating with broken config.
    """
    src_dir = workflow_source(one_conf)
    workflow_id = install(src_dir)

    # Check that the original installation validates OK:
    validate(workflow_id, against_source=True)

    # Break the source config:
    with open(src_dir / 'flow.cylc', 'a') as handle:
        handle.write('\n[runtime]\n[[foo]]\nAgrajag = bowl of petunias')

    # # Check that Validate now fails:
    with pytest.raises(IllegalItemError, match='Agrajag'):
        validate(workflow_id, against_source=True)


async def test_validate_from_source_tvars(
```

Reply:
Resolved locally